actors/miner/tests/util.rs

```diff
@@ -1110,17 +1112,185 @@ impl ActorHarness {
+            });
+            sector_allocation_claims.push(claims);
+
+            let notifications = notifications_from_pieces(&sup.pieces);
+            if !notifications.is_empty() {
+                expected_sector_notifications.push(SectorChanges {
+                    sector: sup.sector,
+                    minimum_commitment_epoch: sector.expiration,
+                    added: notifications,
+                });
+            }
+
+            let duration = sector.expiration - *rt.epoch.borrow();
+            let mut deal_size = DealWeight::zero();
+            let mut verified_size = DealWeight::zero();
+            for piece in &sup.pieces {
+                if piece.verified_allocation_key.is_some() {
+                    verified_size += piece.size.0 * duration as u64;
+                } else {
+                    deal_size += piece.size.0 * duration as u64;
+                }
             }
+
+            let qa_power_delta =
+                qa_power_for_weight(self.sector_size, duration, &deal_size, &verified_size)
+                    - qa_power_for_sector(self.sector_size, &sector);
+            expected_qa_power += &qa_power_delta;
+            let pledge = initial_pledge_for_power(
+                &qa_power_delta,
+                &self.baseline_power,
+                &self.epoch_reward_smooth,
+                &self.epoch_qa_power_smooth,
+                &rt.total_fil_circ_supply(),
+            );
+            expected_pledge += pledge;
+        }
+
+        // Expect claiming of verified space for each piece that specified an allocation ID.
+        if !sector_allocation_claims.iter().all(|sector| sector.claims.is_empty()) {
+            rt.expect_send_simple(
+                VERIFIED_REGISTRY_ACTOR_ADDR,
+                VerifregMethod::ClaimAllocations as u64,
+                IpldBlock::serialize_cbor(&ClaimAllocationsParams {
+                    sectors: sector_allocation_claims,
+                    all_or_nothing: require_activation_success,
+                })
+                .unwrap(),
+                TokenAmount::zero(),
+                IpldBlock::serialize_cbor(&ClaimAllocationsReturn {
+                    sector_results: BatchReturn::ok(sector_updates.len() as u32),
+                    sector_claims: sector_claimed_space,
+                })
+                .unwrap(),
+                ExitCode::OK,
+            );
         }
+
+        // Expect pledge & power updates.
+        self.expect_query_network_info(rt);
+        expect_update_pledge(rt, &expected_pledge);
+        expect_update_power(rt, PowerPair::new(BigInt::zero(), expected_qa_power));
+
+        // Expect SectorContentChanged notification to market.
+        let sector_notification_resps = expected_sector_notifications
+            .iter()
+            .map(|sn| SectorReturn { added: vec![PieceReturn { accepted: true }; sn.added.len()] })
+            .collect();
+        rt.expect_send_simple(
```

Comment:
I think you want an if statement here like the one above so you don't send a message if there are no notifications

Reply:
Thanks, I was missing an `rt.verify()` to catch this.